tests/Unit/Schema/Directives/UploadDirectiveTest.php

```diff
@@ -0,0 +1,300 @@
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => []]
+        )->assertJson([
+            'data' => [
+                'file' => $filePath,
+            ],
+        ]);
+    }
+
+    public function testThrowsAnExceptionIfDiskIsMissing(): void
+    {
+        $this->expectException(Exception::class);
+
+        config()->offsetSet('filesystems.default', 'uploadDisk');
+
+        $this->schema = /** @lang GraphQL */
+            '
+        scalar Upload @scalar(class: "Nuwave\\\\Lighthouse\\\\Schema\\\\Types\\\\Scalars\\\\Upload")
+    
+        type Mutation {
+            upload(
+            file: Upload! @upload(disk:"uploadDisk")
+            ): Boolean
+        }
+        ' . self::PLACEHOLDER_QUERY;
+
+        $file = UploadedFile::fake()->create('test.pdf', 500);
+
+        $this->multipartGraphQL(
+            [
+                'query' => /** @lang GraphQL */ '
+                mutation ($file: Upload!) {
+                    file: upload(file: $file)
+                }
+            ',
+                'variables' => [
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => $file]
+        );
+    }
+
+    public function testThrowsAnExceptionIfStoringFails(): void
+    {
+        $this->expectExceptionObject(new CannotWriteFileException("Unable to upload `upload` file to `/` via disk `uploadDisk`"));
+
+        config()->offsetSet('filesystems.default', 'uploadDisk');
+
+        $filePath = null;
+
+        Storage::fake('uploadDisk');
+
+        $this->schema = /** @lang GraphQL */
+            '
+        scalar Upload @scalar(class: "Nuwave\\\\Lighthouse\\\\Schema\\\\Types\\\\Scalars\\\\Upload")
+    
+        type Mutation {
+            upload(
+            file: Upload! @upload(path:"/")
+            ): String
+        }
+        ' . self::PLACEHOLDER_QUERY;
+
+        $file = Mockery::mock(File::class);
+
+        $file->shouldReceive('hashName')
+            ->andReturn('testFileName.pdf');
+
+        $file->shouldReceive('storeAs')
+            ->andReturn(false);
+
+        $this->multipartGraphQL(
+            [
+                'query' => /** @lang GraphQL */ '
+                mutation ($file: Upload!) {
+                    file: upload(file: $file)
+                }
+            ',
+                'variables' => [
+                    'file' => null,
+                ],
+            ],
+            ['0' => ['variables.file']],
+            ['0' => $file]
+        )->assertJson([
+            'data' => [
+                'file' => $filePath,
+            ],
+        ]);
+
+        Storage::disk('uploadDisk')->assertExists($file->hashName());
+    }
+
+    public function testThrowsAnExceptionIfAttributeIsNotUploadedFile(): void
+    {
+        $this->expectExceptionObject(new InvalidArgumentException("Expected UploadedFile from `baz`"));
```

Comment:
Same as the other test, this should be as close as possible to the source of error.